### PR TITLE
Require Node.js 8, add TypeScript definition

### PR DESCRIPTION
**Pull request by BendingBender**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	 */
+	readonly customReplacements?: ReadonlyArray<[string, string]>;
+}
+
+/**
+ * Slugify a string.
+ *
+ * @param input - The string to slugify.
+ */
+declare function slugify(input: string, options?: Options): string;
```
Comment (sindresorhus) [inline index.d.ts]:
```suggestion
export default function slugify(input: string, options?: Options): string;
```

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
+export interface Options {
+	/**
+	 * @default "-"
```
Comment (sindresorhus) [inline index.d.ts]:
```suggestion
	 * @default '-'
```

File: index.test-d.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import {expectType} from 'tsd-check';
+import {slugify} from '.';
```
Comment (sindresorhus) [inline index.test-d.ts]:
```suggestion
import slugify from '.';
```

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+	 *
+	 * @example
+	 *
```
Comment (sindresorhus) [inline index.d.ts]: I think all the examples should be copy-pastable, meaning we need to include the import statement in each example too.

Comment (BendingBender) [reply index.d.ts]: In the readme.md, too?

Comment (sindresorhus) [reply index.d.ts]: Yes :)

**Pull request merged**